runtime/criusupport/criusupport.cpp

```diff
@@ -262,9 +279,13 @@ toggleSuspendOnJavaThreads(J9VMThread *currentThread, BOOLEAN suspend)
 		&& (currentThread != walkThread)
 		) {
 			if (suspend) {
-				vmFuncs->setHaltFlag(walkThread, J9_PUBLIC_FLAGS_HALT_THREAD_FOR_CHECKPOINT);
+				if (!jdwpSpecialAction || (jdwpSpecialAction && !isJdwpDebugThread(walkThread))) {
```

Comment:
you can define `bool skipThread = ...` above to avoid code duplication

Reply:
Added a boolean, but changed the logic a bit so that the function will only set or clear bits for JDWP threads or non JDWP threads.